mempool/src/shared_mempool/coordinator.rs

```diff
@@ -103,7 +110,25 @@ pub(crate) async fn coordinator<NetworkClient, TransactionValidator>(
                 tasks::execute_broadcast(peer, backoff, &mut smp, &mut scheduled_broadcasts, executor.clone()).await;
             },
             (network_id, event) = events.select_next_some() => {
-                handle_network_event(&executor, &bounded_executor, &mut scheduled_broadcasts, &mut smp, network_id, event).await;
+                handle_network_event(&bounded_executor, &mut smp, network_id, event).await;
+            },
+            _ = update_peers_interval.tick().fuse() => {
+                if let Ok(connected_peers) = peers_and_metadata.get_connected_peers_and_metadata() {
```

Comment:
nit? every other branch of this `select!` has been pulled out into a helper function so that it's a one-liner here